lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,172 @@
+        next unless deployment_node.vm_id.nil?
+        deployment_node.add_vm vm_id
+      end
+    end
+
+    def rhsm_creds
+      [rhsm_auth.userid, rhsm_auth.password]
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      object_send(:add_deployment_provider, options)
+    end
+
+    def regenerate_ansible_inventory
+      object_send(:generate_ansible_yaml)
+    end
+
+    def regenerate_ansible_subscription_inventory
+      object_send(:generate_ansible_inventory_for_subscription)
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        wrap_results(@object.automation_task = AutomationTask.find_by_id(task.id))
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        options.each do |key, val|
+          wrap_results(@object.customizations[:agent][key] = val)
+        end
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip,
+                       username,
+                       local_path,
+                       remote_path,
+                       :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      pp commands
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      result = ssh.perform_commands(commands)
+      result
+    end
+
+    def check_connection(ip, username, ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
+      success, unreachable_ips = agent.with_service do |socket|
+        ssh = LinuxAdmin::SSH.new(ip, username)
+        unreachable_ips = []
+        conneted = true
+        ips.each do |sub_ip|
+          res = ssh.perform_commands(["sudo -E ssh -o 'StrictHostKeyChecking no' #{username}@#{sub_ip} echo $?"], socket)[:stdout]
+          unless res.include?("0\r\n")
+            conneted = false
+            unreachable_ips << sub_ip
+          end
+        end
+        [conneted, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      playbook_pid = ar_method do
+        @object.customizations[:agent][:playbook_pid] if @object.customizations[:agent]
+      end
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      if playbook_pid
+        process_running = ssh.perform_commands(["sudo kill -0 #{playbook_pid}"])[:stdout]
+        unless process_running.empty?
```

Comment:
Double negative on top of being not obivous when kill's stdout is empty or not.
Can you check exist status instead of stdout?
Or can you rename `process_running` to make it obvious what being empty implies?